src/Tribe/Views/V2/Widgets/Admin_Template.php

```diff
@@ -27,10 +49,153 @@ public function __construct() {
 		$this->set_template_origin( tribe( 'tec.main' ) );
 		$this->set_template_folder( 'src/admin-views' );
 
+		/**
+		 * Allows other plugins to add new/custom field types.
+		 *
+		 * @since TBD
+		 *
+		 * @param array<string> An array of field-type "slugs".
+		 */
+		$this->allowed_field_types = apply_filters( 'tribe_events_views_v2_widget_allowed_field_types', $this->allowed_field_types );
+
 		// We specifically don't want to look up template files here.
 		$this->set_template_folder_lookup( false );
 
 		// Configures this templating class extract variables.
 		$this->set_template_context_extract( true );
 	}
+
+	/**
+	 * Sets global widget object and filters out any unexpected data before passing to the conditionals.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_Widget           $widget_obj   The widget object.
+	 * @param array<string,mixed> $admin_fields The array of field(s) data.
+	 * @return void
+	 */
+	public function structure( $widget_obj, $admin_fields ) {
```

Comment:
Not suggesting a change here, but the name `structure` for this method is a bit vague.
Since the method will echo to the page, add something like `print` to it.
And something about `form`, maybe?
My alternative proposal is `printForm`.